File: backend/app/schemas/specialist.py
from pydantic import BaseModel, EmailStr, Field
from typing import Optional, List
from datetime import datetime
from app.models.specialist import Specialty


class SpecialistCreate(BaseModel):
    specialty: Specialty
    subspecialties: List[str] = Field(default_factory=list)
    institution_id: int
    phone: Optional[str] = Field(None, max_length=20)
    email: Optional[EmailStr] = None
    bio: Optional[str] = None
    interests: List[str] = Field(default_factory=list)
    is_public: bool = True


class SpecialistUpdate(BaseModel):
    specialty: Optional[Specialty] = None
    subspecialties: Optional[List[str]] = None
    institution_id: Optional[int] = None
    phone: Optional[str] = Field(None, max_length=20)
    email: Optional[EmailStr] = None
    bio: Optional[str] = None
    interests: Optional[List[str]] = None
    is_public: Optional[bool] = None


class SpecialistUserOut(BaseModel):
    id: int
    name: str
    email: str
    role: str
    department: Optional[str] = None
    phone: Optional[str] = None

    class Config:
        from_attributes = True


class SpecialistFacilityOut(BaseModel):
    id: int
    facility_name: str
    city: str
    region: str
    facility_type: str

    class Config:
        from_attributes = True


class SpecialistOut(BaseModel):
    id: int
    user_id: int
    specialty: Specialty
    subspecialties: List[str]
    institution_id: int
    phone: Optional[str] = None
    email: Optional[str] = None
    bio: Optional[str] = None
    interests: List[str]
    is_public: bool
    created_at: datetime
    updated_at: Optional[datetime] = None
    user: Optional[SpecialistUserOut] = None
    institution: Optional[SpecialistFacilityOut] = None

    class Config:
        from_attributes = True


class SpecialistDetail(SpecialistOut):
    pass

File: frontend/types/specialist.ts
export type Specialty = "GASTROENTEROLOGY" | "HEPATOLOGY" | "GI_SURGERY";

export interface SpecialistUser {
  id: number;
  name: string;
  email: string;
  role: string;
  department?: string;
  phone?: string;
}

export interface SpecialistFacility {
  id: number;
  facility_name: string;
  city: string;
  region: string;
  facility_type: string;
}

export interface Specialist {
  id: number;
  user_id: number;
  specialty: Specialty;
  subspecialties: string[];
  institution_id: number;
  phone?: string;
  email?: string;
  bio?: string;
  interests: string[];
  is_public: boolean;
  created_at: string;
  updated_at?: string;
  user?: SpecialistUser;
  institution?: SpecialistFacility;
}

export const SPECIALTY_LABELS: Record<Specialty, string> = {
  GASTROENTEROLOGY: "Gastroenterology",
  HEPATOLOGY: "Hepatology",
  GI_SURGERY: "GI Surgery",
};

export const COMMON_SUBSPECIALTIES = [
  "Advanced Endoscopy",
  "Inflammatory Bowel Disease",
  "Transplant Hepatology",
  "Viral Hepatitis",
  "Motility Disorders",
  "Nutrition",
  "Paediatric GI",
];

export const COMMON_INTERESTS = [
  "Colonoscopy",
  "ERCP",
  "EUS",
  "IBD",
  "Hepatitis B",
  "Hepatitis C",
  "Cirrhosis",
  "HCC surveillance",
  "UGIB",
  "Coeliac Disease",
];
